critiquebrainz/db/avg_rating.py

```diff
@@ -0,0 +1,152 @@
+	with db.engine.connect() as conenction:
+		result = conenction.execute(sqlalchemy.text("""
+			SELECT count(*)
+			  FROM avg_rating
+			 WHERE entity_id = :entity_id
+			   AND entity_type = :entity_type
+		"""),{
+			"entity_id": entity_id,
+			"entity_type": entity_type,
+		})
+
+		count = result.fetchone()[0]
+		if count:
+			update(entity_id, entity_type)
+		else
+			create(entity_id, entity_type, rating)
+
+def create(entity_id, entity_type, rating):
+	"""Inserts rating and count for given entity
+	
+	Args:
+		entity_id (uuid): ID of the entity rated
+		entity_type (str): Type of the entity rated
+		rating (int): Rating of the entity
+	"""
+	with db.engine.connect() as connection:
+		connection.execute(sqlalchemy.text("""
+			INSERT INTO avg_rating(entity_id, entity_type, rating, count)
+				 VALUES (:entity_id, :entity_type, :rating, :count)
+			"""), {
+				"entity_id": entity_id,
+				"entity_type": entity_type,
+				"rating": rating,
+				"count": 1,
+			})
+
+
+def update(entity_id, entity_type):
+	"""Updates the average rating of the entity
+	
+	It does so by collecting all the review_ids for a given entity from review
+	and selecting respective revisions with latest timestamp
+	and calculating sum and count of ratings 
+	
+	Args:
+		entity_id (uuid): ID of the entity
+		entity_type (str): Type of the entity
+	"""
+	with db.engine.connect() as connection:
+		result = connection.execute(sqlalchemy.text("""
+			WITH LatestRevisions AS (
+				SELECT review_id,
+					   MAX("timestamp") created_at
+				  FROM revision
+				 WHERE review_id in ( 
+				 	       SELECT id 
+				 	         FROM review
+				 	        WHERE entity_id = :entity_id
+				 	          AND entity_type = :entity_type)
+			  GROUP BY review_id)
+			
+			SELECT SUM(rating),
+			       COUNT(rating)
+			  FROM revision
+		INNER JOIN LatestRevisions
+		        ON revision.review_id = LatestRevisions.review_id
+		       AND revision.timestamp = LatestRevisions.created_at
+		"""),{
+			"entity_id": entity_id,
+			"entity_type": entity_type,
+		})
+
+		row = result.fetchone()
+		if row is None:
+			delete(entity_id)
+			return
+	
+	#Calulate average rating and update it	
+	sum, count = row[0], row[1]
+	avg_rating = int(sum / count + 0.5)
+	with db.engine.connect() as connection:
+		connection.execute(sqlalchemy.text("""
+			UPDATE avg_rating
+			   SET rating = :avg_rating,
+			       count = :count
+			 WHERE entity_id = :entity_id
+			   AND entity_type = :entity_type
+	    """),{
+			"avg_rating": avg_rating,
+			"count": count,
+			"entity_id": entity_id,
+			"entity_type": entity_type,
+	    })
+
+def delete(entity_id):
+	"""Deletes the avg_rating, given entity_id and entity_type
+
+	Args: 
+		entity_id (uuid): ID of the entity
+	"""
+	with db.engine.connect() as connection:
+		connection.execute(sqlalchemy.text("""
+			DELETE 
+			  FROM avg_rating
+			 WHERE entity_id = :entity_id
+		"""), {
+			"entity_id": entity_id,
+		})
+
+def get_avg_rating(entity_id):
```

Comment:
It would be better to name this function to simply `get` to be consistent with the rest in this module. It shouldn't be confusing what's being retrieved since it's in the name of the module.